wafer/schedule/admin.py

```diff
@@ -106,6 +106,29 @@ def find_invalid_venues(all_items):
     return venues.items()
 
 
+def find_speaker_clashes(all_items):
+    """Find items that have the same speaker and also have overlapping
+       slots"""
+    clashes = {}
+    seen_slots_speakers = {}
+    for item in all_items:
+        if item.talk:
+            speakers = list(item.talk.authors.all())
+        elif item.page:
+            speakers = list(item.page.people.all())
+        for speaker in speakers:
+            for slot in item.slots.all():
+                candidate = (slot, speaker)
+                if candidate in seen_slots_speakers:
+                    if seen_slots_speakers[candidate] not in clashes:
+                        clashes[candidate] = [seen_slots_speakers[candidate]]
+                    clashes[candidate].append(item)
```

Comment:
Is it possible to return which speaker has the clash? I guess in many cases it will be obvious but it might be easier for the user to sort out if we can display the extra information.

Reply:
The error message already does include that, because it's built up from the returned items which includes those details (although the formatting could be improved)

```Common speaker in simultaneous schedule items: ((<Slot: Slot: Oct 14 (Fri): 10:00 - 11:00>, <User: neil>), [<ScheduleItem: T3est in Venue1 at Oct 14 (Fri): 10:00>, <ScheduleItem: tt in Venue2 at Oct 14 (Fri): 10:00>])```

Reply:
I've now improved the display in the admin page

Reply:
Thanks!